src/Kestrel.Transport.Sockets/Internal/SocketConnection.cs

```diff
@@ -244,7 +244,14 @@ public void Dispose()
 
                 // This is not interlocked because there could be a concurrent writer.
                 // Instead it's to prevent read tearing on 32-bit systems.
-                Interlocked.Add(ref _totalBytesWritten, buffer.Length);
+                if (IntPtr.Size == sizeof(long))
+                {
+                    _totalBytesWritten += buffer.Length;
+                }
+                else
+                {
+                    Interlocked.Add(ref _totalBytesWritten, buffer.Length);
```

Comment:
Here a variant with `Volatile` isn't as efficient as this one.

```c#
long totalBytesWritten = Volatile.Read(ref _totalBytesWritten);
totalBytesWritten += buffer.Length;
Volatile.Write(ref _totalBytesWritten, totalBytesWritten);
```

produces

```asm
00007FFC0DCA1570  lea         rax,[rcx+8]  
00007FFC0DCA1574  mov         rcx,rax  
00007FFC0DCA1577  mov         rcx,qword ptr [rcx]  
00007FFC0DCA157A  add         rcx,rdx  
00007FFC0DCA157D  mov         qword ptr [rax],rcx  
```

instead of a simple

```asm
00007FFC0DCA1550  add         qword ptr [rcx+8],rdx  
```
(for the x64-path).

|         Method |     Mean |     Error |    StdDev | Scaled |
|--------------- |---------:|----------:|----------:|-------:|
| InterlockedAdd | 6.159 ns | 0.0545 ns | 0.0510 ns |   1.00 |
| DirectAddOnx64 | 1.010 ns | 0.0319 ns | 0.0298 ns |   0.16 |
|    VolatileAdd | 1.354 ns | 0.0282 ns | 0.0264 ns |   0.22 |

So this one should be kept as-is in this PR.